integration-cli/docker_cli_pull_test.go

```diff
@@ -272,3 +272,10 @@ func (s *DockerSuite) TestPullLinuxImageFailsOnWindows(c *check.C) {
 	_, _, err := dockerCmdWithError("pull", "ubuntu")
 	c.Assert(err.Error(), checker.Contains, "cannot be used on this platform")
 }
+
+// Regression test for https://github.com/docker/docker/issues/28892
+func (s *DockerSuite) TestPullWindowsImageFailsOnLinux(c *check.C) {
+	testRequires(c, DaemonIsLinux, Network)
+	_, _, err := dockerCmdWithError("pull", "microsoft/nanoserver")
```

Comment:
Ideally we'd serve a minimal Windows image from a local registry instead of depending on Docker Hub. I just filed https://github.com/docker/docker/issues/30626 about something similar.

This is probably outside the scope of this PR though. I think the `download-frozen-image-v2.sh` script in the `Dockerfile` is capable of downloading the image at `docker build` time, but it would take some extra scripting work to load this image into a local registry without the use of `docker pull` / `docker push`.